internal/controllers/cluster/cluster_controller.go

```diff
@@ -62,19 +62,19 @@ const (
 // +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=clusters;clusters/status;clusters/finalizers,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=apiextensions.k8s.io,resources=customresourcedefinitions,verbs=get;list;watch
 
-// ClusterReconciler reconciles a Cluster object.
-type ClusterReconciler struct {
+// Reconciler reconciles a Cluster object.
+type Reconciler struct {
 	Client    client.Client
 	APIReader client.Reader
 
 	// WatchFilterValue is the label value used to filter events prior to reconciliation.
 	WatchFilterValue string
 
-	recorder        record.EventRecorder
+	Recorder        record.EventRecorder
```

Comment:
why is this being exported?

Reply:
This is required because machine controller is used by test suites in other packages, e.g the test suite in internals/controllers/cluster, and this requires being able to pass a fake recorder

Reply:
>because machine controller is used by test suites in other packages,

That should be the case only for envtest in which case a real recorder should be always used right (which also makes me wonder if we should envtest through the alias)? for units which might need fake recorder they should always be in the same package right? just curious I think exporting is also ok anyways.

Reply:
Related: https://github.com/kubernetes-sigs/cluster-api/pull/5899#discussion_r778868788
(maybe we don't have to export it)

Reply:
fixed.